### PR TITLE
add star schema recipe

### PR DESCRIPTION
**Pull request by knaufk**

### CODE REVIEW

File: recipes/19/19_start_schema.md [hunk 1]
```diff
@@ -0,0 +1,230 @@
+# 16 Real Time Star Schema Denormalization
```
Comment (morsapaes) [inline recipes/19/19_start_schema.md:1]: I'd call it simply "N-Way Joins".

File: recipes/19/19_start_schema.md [hunk 2]
```diff
@@ -0,0 +1,230 @@
+# 16 Real Time Star Schema Denormalization 
+
+:bulb: In this recipe we will de-normalize a simple star schema with an n-way temporal table join. 
+
+Our small data warehouse consists of a fact table (`train_activity`) and three dimension tables (`stations`, `booking_channels` and `passengers`). 
+All inserts to the fact table and all updates to the dimension tables are mirrored to Apache Kafka.
+Hence, all our tables are backed by Kafka topics, but in the case of rates this topic is compacted (i.e. only the most recent messages for a given key are kept as updated rates flow in).
```
Comment (morsapaes) [inline recipes/19/19_start_schema.md:7]: "Rates"?

File: recipes/19/19_start_schema.md [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+Records in the fact table are interpreted as inserts only, and so the table is backed by the [standard Kafka connector](https://ci.apache.org/projects/flink/flink-docs-stable/dev/table/connectors/kafka.html) (`connector` = `kafka`); while the records in the dimensional tables need to be interpreted as upserts based on a primary key, which requires the [Upsert Kafka connector](https://ci.apache.org/projects/flink/flink-docs-stable/dev/table/connectors/upsert-kafka.html) (`connector` = `upsert-kafka`).
+
+With Flink SQL we can now easily join all dimensions to our fact table using a 5-way temporal table join. 
+Temporal table joins take an arbitrary table (left input/probe site) and correlate each row to the corresponding row’s relevant version in a versioned table (right input/build side). 
+Flink uses the SQL syntax of ``FOR SYSTEM_TIME AS OF`` to perform this operation. 
+Using a temporal table join leads to consistent, reproducible results when joining a fact table with one are more (slowly) changing dimensional tables.
```
Comment (morsapaes) [inline recipes/19/19_start_schema.md:13]:
```suggestion
Using a temporal table join leads to consistent, reproducible results when joining a fact table with one or more (slowly) changing dimension tables. 
```

File: recipes/19/19_start_schema.md [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+Records in the fact table are interpreted as inserts only, and so the table is backed by the [standard Kafka connector](https://ci.apache.org/projects/flink/flink-docs-stable/dev/table/connectors/kafka.html) (`connector` = `kafka`); while the records in the dimensional tables need to be interpreted as upserts based on a primary key, which requires the [Upsert Kafka connector](https://ci.apache.org/projects/flink/flink-docs-stable/dev/table/connectors/upsert-kafka.html) (`connector` = `upsert-kafka`).
+
+With Flink SQL we can now easily join all dimensions to our fact table using a 5-way temporal table join. 
+Temporal table joins take an arbitrary table (left input/probe site) and correlate each row to the corresponding row’s relevant version in a versioned table (right input/build side). 
+Flink uses the SQL syntax of ``FOR SYSTEM_TIME AS OF`` to perform this operation. 
+Using a temporal table join leads to consistent, reproducible results when joining a fact table with one are more (slowly) changing dimensional tables. 
+Every event (row in the fact table) is joined to its corresponding value of each dimension as of the time when the event.
```
Comment (morsapaes) [inline recipes/19/19_start_schema.md:14]:
```suggestion
Every event (i.e. row in the fact table) is joined to its corresponding value in each dimension as of the time when the event happened.
```

File: recipes/19/19_start_schema.md [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+### JobGraph
+
+![jobgraph](19_jobgraph.png)
+
+## Data Generators
+
+<details>
+    <summary>Data Generators</summary>
+
+The four topics are populated Flink SQL jobs, too.
```
Comment (morsapaes) [inline recipes/19/19_start_schema.md:111]:
```suggestion
The four topics are populated with Flink SQL jobs, too.
```

File: recipes/19/19_start_schema.md (hunk 4 above)
Comment (sjwiesman) [inline recipes/19/19_start_schema.md:14]:
```suggestion
bulb: In this recipe, we will de-normalize a simple star schema with an n-way temporal table join. 	 
  
[Star schemas](https://en.wikipedia.org/wiki/Star_schema) are a popular way of normalizing data within a data warehouse. At the center of a star schema is a **fact table** whose rows contain metrics, measurements, and other facts about the world. Surrounding fact tables are one or more **dimension tables** which have metadata useful for enriching facts when computing queries.  
You are running a small data warehouse for a railroad company which consists of a fact table (`train_activity`) and three dimension tables (`stations`, `booking_channels`, and `passengers`). All inserts to the fact table, and all updates to the dimension tables, are mirrored to Apache Kafka. 
Records in the fact table are interpreted as inserts only, and so the table is backed by the [standard Kafka connector](https://ci.apache.org/projects/flink/flink-docs-stable/dev/table/connectors/kafka.html) (`connector` = `kafka`);. In contrast, the records in the dimensional tables are upserts based on a primary key, which requires the [Upsert Kafka connector](https://ci.apache.org/projects/flink/flink-docs-stable/dev/table/connectors/upsert-kafka.html) (`connector` = `upsert-kafka`).	 

With Flink SQL you can now easily join all dimensions to our fact table using a 5-way temporal table join. 	 
Temporal table joins take an arbitrary table (left input/probe site) and correlate each row to the corresponding row’s relevant version in a versioned table (right input/build side). 	 
Flink uses the SQL syntax of ``FOR SYSTEM_TIME AS OF`` to perform this operation. 	 
Using a temporal table join leads to consistent, reproducible results when joining a fact table with more (slowly) changing dimensional tables. 	 
 Every event (row in the fact table) joins to its corresponding value of each dimension based on when the event occurred in the real world. 
```

**Pull request merged**